### PR TITLE
CPB-(30-31): users' profile List/Read/Update/Delete api

### PR DESCRIPTION
**Pull request by devoorhees**

### CODE REVIEW

File: backend/apps/accounts/api/v1/filters.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    first_name = django_filters.CharFilter(
+        field_name="user_profile__first_name", lookup_expr="icontains"
+    )
+    last_name = django_filters.CharFilter(
+        field_name="user_profile__last_name", lookup_expr="icontains"
+    )
+
+    class Meta:
+        model = UserModel
+        fields = ["email", "first_name", "last_name"]
```
Comment (pazuzu-ua) [inline backend/apps/accounts/api/v1/filters.py]: Consider using a tuple here

File: backend/apps/accounts/api/v1/views.py [hunk 2]
```diff
@@ -100,3 +110,87 @@ def post(self, request, *args, **kwargs):
...
+    serializer_class = UserListSerializer
+    pagination_class = UserListPagination
+    permission_classes = [permissions.IsAuthenticated]
+    filter_backends = [
+        django_filters.DjangoFilterBackend,
+        filters.SearchFilter,
+        filters.OrderingFilter,
+    ]
+    filterset_class = UserFilter
+    search_fields = ["=email", "user_profile__first_name", "user_profile__last_name"]
```
Comment (pazuzu-ua) [inline backend/apps/accounts/api/v1/views.py]: Consider using tuples here

File: backend/apps/accounts/api/v1/views.py [hunk 3]
```diff
@@ -100,3 +110,87 @@ def post(self, request, *args, **kwargs):
...
+    ordering_fields = ["email", "first_name", "last_name"]
+    ordering = ["first_name", "last_name"]
+
+    def get_queryset(self):
+        """
+        Annotate a queryset to ba able to use user's `first_name` and `last_name`
```
Comment (pazuzu-ua) [inline backend/apps/accounts/api/v1/views.py]: to b**e** able

File: backend/apps/accounts/api/v1/views.py [hunk 4]
```diff
@@ -100,3 +110,87 @@ def post(self, request, *args, **kwargs):
...
+        """Partially update User by id."""
+        return self.partial_update(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        """Don't actually delete the user, set User account as inactive instead."""
+        return self.destroy(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        instance.is_active = False
+        instance.save()
```
Comment (pazuzu-ua) [inline backend/apps/accounts/api/v1/views.py]: Use update_fields maybe?

File: backend/apps/accounts/api/v1/views.py [hunk 5]
```diff
@@ -100,3 +110,87 @@ def post(self, request, *args, **kwargs):
...
+    patch <id>:
+        Partially update a User with its Profile info.
+
+    delete <id>:
+        Deactivate the user account without complete deletion.
+
+    """
+
+    serializer_class = UserRetrieveUpdateDestroySerializer
+    permission_classes = [permissions.IsAuthenticated, IsUserAccount]
```
Comment (pazuzu-ua) [inline backend/apps/accounts/api/v1/views.py]: Consider using tuples here

File: backend/apps/accounts/tests/test_user_listview.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_authorized_user_get_userlist(self):
+        """Ensure authorized user can get a user list."""
+
+        self.client.force_authenticate(user=self.user)
+        response = self.client.get(self.user_list_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json().get("count"), 5)
```
Comment (pazuzu-ua) [inline backend/apps/accounts/tests/test_user_listview.py]: or you can use `response.data.get("count")`

**Review by pazuzu-ua — APPROVED** [review]

**Review by LV-coding — APPROVED** [review]

**Pull request merged**